vars/githubscm.groovy

```diff
@@ -45,21 +45,28 @@ def getRepositoryScm(String repository, String author, String branches, String c
 }
 
 def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceBranches, String targetAuthor, String targetBranches, String credentialId = 'kie-ci') {
-    println "[INFO] Merging source [${sourceAuthor}/${repository}:${sourceBranches}] into target [${targetAuthor}/${repository}:${targetBranches}]..."
+    def sourceRepository = repository
+    try {
+        sourceRepository = util.getSourceProjectGroupName()[1]
+    } catch (Exception) {
+        println "[WARNING] can't take source project name for '${repository}'. Taking target as source"
+    }
+
+    println "[INFO] Merging source [${sourceAuthor}/${sourceRepository}:${sourceBranches}] into target [${targetAuthor}/${repository}:${targetBranches}]..."
     checkout(resolveRepository(repository, targetAuthor, targetBranches, false, credentialId))
     def targetCommit = getCommit()
 
     try {
         withCredentials([usernameColonPassword(credentialsId: credentialId, variable: 'kieCiUserPassword')]) {
-            sh "git pull https://${kieCiUserPassword}@github.com/${sourceAuthor}/${repository} ${sourceBranches}"
+            sh "git pull https://${kieCiUserPassword}@github.com/${sourceAuthor}/${sourceRepository} ${sourceBranches}"
```

Comment:
@Ginxo In case you're not aware of this option, you can obtain the merge commit of source and target branches directly by fetching this:
```
$ git fetch origin pull/ID/merge:BRANCHNAME
```
Notice the *merge* part. More info in GitHub [docs](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally) but surprisingly they only mention the *head* variant, which is the state of the source branch in the fork (not what we want to build on Jenkins). But I've been using `pull/ID/merge` to check PRs locally for a long time. It works.

Mentioning this as it could be used to simplify the script.